### ISSUE TITLE
BSD detect_cmake: _INCLUDED_FILE "NOTFOUND"

### ISSUE THREAD

**Issue by ajacoutot** [mesonbuild/meson#5218]
Hi.

On at least OpenBSD and FreeBSD, the cmake module fails to determine the compilerpath. This is because it looks at the _INCLUDED_FILE value:
```
$ cmake --system-information -G Ninja | grep _INCLUDED_FILE
```
On BSD, the output is:
```
_INCLUDED_FILE "NOTFOUND"
```
On Linux, the output is:
```
_INCLUDED_FILE "/usr/share/cmake-3.12/Modules/Platform/Linux-GNU-CXX.cmake"
```

I wonder is we could maybe use something like _INCLUDED_SYSTEM_INFO_FILE which has the following output (according to the OS):
```
/usr/local/share/cmake/Modules/Platform/OpenBSD.cmake
/usr/share/cmake-3.12/Modules/Platform/Linux.cmake"
```

**Comment by dcbaker** [comment]
@mensinda, you're the most knowledgable about the cmake backend I think.

**Comment by mensinda** [comment]
Is this even an issue? I mean does the dependency detection fail because of this? The CMake dependency should avoid the entire compiler detection by presenting CMake with a "fake" build directory.

Is there an easy way to reproduce the error (I have never worked with any BSD)?

**Comment by ajacoutot** [comment]
@mensinda I am not sure whether this has real world consequences. I run into this when running the meson regress tests on BSD.
To reproduce on Linux, just edit the CmakeModule class in mesonbuild/modules/cmake.py and replace _INCLUDED_FILE with some random garbage (e.g. _FOOBAR).

**Comment by mensinda** [comment]
@ajacoutot I thought you were talking about the CMake dependency backend... I didn't even know that this module exists. It looks like `_INCLUDED_FILE` can be replaced with `CMAKE_ROOT` anyway, since only the CMake path is stored.

Would you mind posting the results of `cmake --system-information -G Ninja | grep CMAKE_ROOT`? This should work on every platform.

**Comment by ajacoutot** [comment]
Sorry I wasn't clear in my original post @mensinda :-)
Here's the output:
```
$ cmake --system-information -G Ninja | grep CMAKE_ROOT
CMAKE_ROOT "/usr/local/share/cmake"
```

**Comment by mensinda** [comment]
Looks good to me, it should be possible to change it to `CMAKE_ROOT` then. It looks like this module was written by @hardening. Were there any special reasons for choosing `_INCLUDED_FILE` over `CMAKE_ROOT`?

**Comment by hardening** [comment]
Sorry for the late reply, using `CMAKE_ROOT` looks perfectly fine.

**Comment by mensinda** [comment]
Should be fixed with #5324
